docs/DafnyRef/Topics.md

```diff
@@ -1146,3 +1146,24 @@ method TestD(dd: D) {
+
+Besides not finding triggers, another problematic situation is when Dafny was able to generate triggers, but believes the triggers it found may still cause useless instantiations because they create matching loops. Dafny emits a warning when this happens, indicating the Dafny program should be changed so Dafny can find triggers for this quantifier that do not cause matching loops.
+
+To understand matching loops, one needs to understand how triggers are used. During verification, Dafny maintains a set of expression which it believes to be true, which we call ground terms. Given a univeral quantifier that's a ground term, Dafny will use its triggers to pattern match on other ground terms. If the pattern matches, that ground term is used to instantiate the quantifier. 
+
+Dafny makes sure not to perform the exact same instantiation twice. However, if an instantiation leads to a new term that also matches the trigger, but is different from the term used for the instantiation, the quantifier may be instantiated too often, which we call a matching loop. For example, given the ground terms `f(3)` and `forall {f(x)} x :: f(x) + f(f(x))`, where `{f(x)}` indicates the trigger for the quantifier, Dafny may instantiate the quantifier using `3` for `x`. This creates a new ground term `f(3) + f(f(3))`, of which the right hand side again matches the trigger, allowing Dafny to instantiate the quantifier again using `f(3)` for `x`, leading to an unbounded amount of instantiations.
+
+Even existential quantifiers need triggers. This is because when Dafny determines an existential quantifier is false, for example in `requires !exists x :: f(x) == 2`, Dafny will use a logical rewrite rule to change this existential into a universal quantifier, so it becomes `requires forall x :: f(x) != 2`. Before verification, Dafny can not determine whether quantifiers will be determined to be true or false, so it must assume any quantifier may turn into a universal quantifier, and thus they all need triggers.
+
+Besides quantifiers, comprehensions such as set and map comprehension also need triggers, since these are modelled using universal quantifiers.
+
+Dafny may report 'Quantifier was split into X parts'. This occurs when Dafny determines it can only generate good triggers for a quantifier by splitting it into multiple smaller quantifiers, whose disjunction is logically equivalent to the original one. To maintain logical equivalence, Dafny has to generate more triggers than if the split had been done in source. One may consider doing the split in source to generate a leaner set of triggers.
```

Comment:
I would love to see an example of that, otherwise that paragraph leaves me with unsatisfied thirst.

For example, you can say `exists x :: P(x) || Q(x)` is split into  `exists x :: P(x)` and `exists x :: Q(x)`. And you can use that example to comment on the trigger generation as well. Also, the disjunction is proper to existential quantifiers, not forall quantifiers. For them, it's a conjunction.